EventFilter/CTPPSRawToDigi/src/CTPPSPixelDataFormatter.cc

```diff
@@ -0,0 +1,183 @@
+  ROC_mask  = ~(~CTPPSPixelDataFormatter::Word32(0) << ROC_bits);    
+
+  maxROCIndex=3; 
+
+  DCOL_mask = ~(~CTPPSPixelDataFormatter::Word32(0) << DCOL_bits);
+  PXID_mask = ~(~CTPPSPixelDataFormatter::Word32(0) << PXID_bits);
+  ADC_mask  = ~(~CTPPSPixelDataFormatter::Word32(0) << ADC_bits);
+
+}
+
+void CTPPSPixelDataFormatter::interpretRawData(  bool& errorsInEvent, int fedId, const FEDRawData& rawData, Collection & digis)
+{
+
+//cout << "Inside interpretRawData" << endl;
+
+  int nWords = rawData.size()/sizeof(Word64);
+  if (nWords==0) return;
+
+// check CRC bit
+  const Word64* trailer = reinterpret_cast<const Word64* >(rawData.data())+(nWords-1);  
+  if(!errorcheck.checkCRC(errorsInEvent, fedId, trailer)) return;
+
+// check headers
+  const Word64* header = reinterpret_cast<const Word64* >(rawData.data()); header--;
+  bool moreHeaders = true;
+  while (moreHeaders) {
+    header++;
+    LogTrace("")<<"HEADER:  " <<  print(*header);
+    bool headerStatus = errorcheck.checkHeader(errorsInEvent, fedId, header);
+    moreHeaders = headerStatus;
+  }
+
+// check trailers
+  bool moreTrailers = true;
+  trailer++;
+  while (moreTrailers) {
+    trailer--;
+    LogTrace("")<<"TRAILER: " <<  print(*trailer);
+    bool trailerStatus = errorcheck.checkTrailer(errorsInEvent, fedId, nWords, trailer);
+    moreTrailers = trailerStatus;
+  }
+
+// data words
+  theWordCounter += 2*(nWords-2);
+  LogTrace("")<<"data words: "<< (trailer-header-1);
+
+  int link = -1;
+  int roc  = -1;
+//  int layer = 0;
+
+  bool skipROC=false;
+
+  edm::DetSet<CTPPSPixelDigi> * detDigis=nullptr;
+
+  const  Word32 * bw =(const  Word32 *)(header+1);
+  const  Word32 * ew =(const  Word32 *)(trailer);
+  if ( *(ew-1) == 0 ) { ew--;  theWordCounter--;}
+  for (auto word = bw; word < ew; ++word) {
+    LogTrace("")<<"DATA: " <<  print(*word);
+
+    auto ww = *word;
+    if unlikely(ww==0) { theWordCounter--; continue;}
+    int nlink = (ww >> LINK_shift) & LINK_mask; 
+    int nroc  = (ww >> ROC_shift) & ROC_mask;
+
+    int FMC = 0;
+
+    int convroc = nroc-1;
+    CTPPSPixelFramePosition fPos(fedId, FMC, nlink, convroc);
+    std::map<CTPPSPixelFramePosition, CTPPSPixelROCInfo>::const_iterator mit;
+    mit = mapping_.find(fPos);
+
+    if (mit == mapping_.end()){      
+      if((nroc-1)>=maxROCIndex){
+	errorcheck.checkROC(errorsInEvent, fedId,  ww); // check kind of error
+      }else{
+	edm::LogError("")<< " CTPPS Pixel DAQ map error " ;
+      }
+      continue; //skip word
+    }
+
+    CTPPSPixelROCInfo rocInfo = (*mit).second;
+
+    CTPPSPixelROC rocp(rocInfo.iD, rocInfo.roc, convroc);
+
+    if ( (nlink!=link) | (nroc!=roc) ) {  // new roc
+      link = nlink; roc=nroc;
+
+      skipROC = likely((roc-1)<maxROCIndex) ? false : !errorcheck.checkROC(errorsInEvent, fedId,  ww); 
+      if (skipROC) continue;
+
+      auto rawId = rocp.rawId();
+//    cout << "+++++++++++++++++++++++++++++ rawId for new ROC  " << rawId << endl;
```

Comment:
is this commented out code needed?
Please remove or add comments inline in the code why the commented out block is relevant